cdap-docs/reference-manual/source/http-restful-api/security.rst

```diff
@@ -11,3 +11,406 @@ Security HTTP RESTful API
 
 .. highlight:: console
 
+Use the Security HTTP RESTful API to manage privileges of users on CDAP entities as well as manage secure storage.
+
+The HTTP RESTful API is divided into these sections:
+
+- :ref:`Authorization <http-restful-api-authorization>`
+- :ref:`Secure Storage <http-restful-api-secure-storage>`
+
+.. Base URL explanation
+.. --------------------
+.. include:: base-url.txt
+
+
+.. _http-restful-api-authorization:
+
+Authorization HTTP RESTful API
+==============================
+
+Use the CDAP Authorization HTTP RESTful API to grant, revoke and list privileges on CDAP entities. Details about
+authorization in CDAP can be found at :ref:`Admin' Manual: Authorization <admin-authorization>`.
+
+.. _http-restful-api-security-auth-grant:
+
+Grant Privileges
+----------------
+You can grant privileges to a principal on a CDAP Entity by making an HTTP POST request to the URL::
+
+  POST /v3/security/authorization/privileges/grant
+
+with JSON-formatted body that contains the principal, the CDAP entity and the actions to be granted::
+
+  {
+	  "entity": {
+		  "namespace": "default",
+		  "dataset": "dataset",
+		  "entity": "DATASET"
+	  },
+	  "principal": {
+		  "name": "alice",
+		  "type": "USER",
+	  },
+	  "actions": ["READ", "WRITE", "ADMIN"]
+  }
+
+
+In the above JSON body, the ``entity`` object is the JSON-serialized form of the CDAP entity classes - e.g. for
+datasets, it is the JSON representation of the `DatasetId
+<https://github.com/caskdata/cdap/blob/release/3.5/cdap-proto/src/main/java/co/cask/cdap/proto/id/DatasetId.java>`__
+class, whereas for programs, it is the JSON representation of the `ProgramId
+<https://github.com/caskdata/cdap/blob/release/3.5/cdap-proto/src/main/java/co/cask/cdap/proto/id/ProgramId.java>`__
+class.
+
+The ``principal`` object refers to the principal that you want to grant the privileges to. Principals have a ``name``
+and a ``type``. The supported types are ``USER``, ``GROUP`` and ``ROLE``. Please note that ``ROLE`` is experimental
```

Comment:
I'm actually thinking we could just remove it. If someone specifically asks for it, we could let them know, but I want to avoid us to have to continue support for ``ROLE`` even after we have support in external systems like Hue.